docs/source/tutorial_worlds.rst

```diff
@@ -0,0 +1,227 @@
+..
+  Copyright (c) 2017-present, Facebook, Inc.
+  All rights reserved.
+  This source code is licensed under the BSD-style license found in the
+  LICENSE file in the root directory of this source tree. An additional grant
+  of patent rights can be found in the PATENTS file in the same directory.
+
+Batching and Hogwild
+====================
+**Author**: Alexander Holden Miller
+
+Summary
+^^^^^^^
+There's one function we need to support for both hogwild and batching: ``share()``.
+
+This function should provide whatever is needed to set up a "copy" of the original
+instance of the agent for either each row of a batch or each thread in hogwild.
+
+The same function is used for both batching and hogwild, since most agents only
+use one or the other. However, an agent may check the ``numthreads`` and/or
+``batchsize`` options to adjust its behavior if it wants to support both, and
+we do support doing both batching and hogwild at the same time if the agent
+desires.
+
+
+Hogwild (multiprocessing)
+^^^^^^^^^^^^^^^^^^^^^^^^^
+Hogwild is initialized in the following way:
+
+1. We set up a starting instance of the world: that is, we use ``create_task``
+   to set up a base world with the appropriate agents and tasks.
+2. We pass this world to a ``HogwildWorld``, which sets up a number of
+   synchronization primitives
+3. We launch ``numthreads`` threads, each initialized from a ``share()``'d
+   version of the world and the agents therein.
+4. Once these threads and their world copies are all launched, we return control back
+
+Now that this world is set up, every time we call parley on it, it will release
```

Comment:
link to code?

Reply:
are you sure? they should see this world as behaving like a regular one I think rather than worrying about the implementation

Reply:
you can just link it, but not actually put the code, so it doesn't take up space, yes i think it's good

Reply:
oh I think the code is very informative! I'd like to keep it